### PR TITLE
Added lhs query alias for joins

### PR DESCRIPTION
**Pull request by benPearce1**
[sc-39691]

### CODE REVIEW

File: source/Nevermore.Tests/QueryBuilderFixture/QueryBuilderFixture.cs [hunk 1]
```diff
@@ -1887,6 +1887,51 @@ public void ShouldGenerateUniqueParameterNamesInJoin()
 
             actual.Should().BeEquivalentTo(expected);
         }
+        
+        [Test]
+        public void ShouldGenerateStaticJoinAliasNamesInJoin()
```
Comment (benPearce1) [inline source/Nevermore.Tests/QueryBuilderFixture/QueryBuilderFixture.cs:1892]: This test is a copy of `ShouldGenerateUniqueParameterNamesInJoin` above it, but with static aliases

File: source/Nevermore/Advanced/QueryBuilders/SubquerySourceBuilder.cs [hunk 2]
```diff
@@ -25,7 +25,7 @@ protected override ISelectBuilder CreateSelectBuilder()
             return new SubquerySelectBuilder(AsSource());
         }
 
-        public override IJoinSourceQueryBuilder<TRecord> Join(IAliasedSelectSource source, JoinType joinType, CommandParameterValues parameterValues, Parameters parameters, ParameterDefaults parameterDefaults)
+        public override IJoinSourceQueryBuilder<TRecord> Join(IAliasedSelectSource source, JoinType joinType, CommandParameterValues parameterValues, Parameters parameters, ParameterDefaults parameterDefaults, string queryAlias = null)
```
Comment (benPearce1) [inline source/Nevermore/Advanced/QueryBuilders/SubquerySourceBuilder.cs:28]: Not sure of the use of the `Join` method here, the `JoinSourceQueryBuilder` purpose seems to be towards setting up the `ON` statement, so the function and the additional `queryAlias` here serve no purpose, unless I am missing something.

File: source/Nevermore/Advanced/QueryBuilders/JoinSourceQueryBuilder.cs [hunk 3]
```diff
@@ -38,7 +38,7 @@ protected override ISelectBuilder CreateSelectBuilder()
             return new JoinSelectBuilder(joinedSource);
         }
 
-        public override IJoinSourceQueryBuilder<TRecord> Join(IAliasedSelectSource source, JoinType joinType, CommandParameterValues parameterValues, Parameters parameters, ParameterDefaults parameterDefaults)
+        public override IJoinSourceQueryBuilder<TRecord> Join(IAliasedSelectSource source, JoinType joinType, CommandParameterValues parameterValues, Parameters parameters, ParameterDefaults parameterDefaults, string queryAlias = null)
```
Comment (benPearce1) [inline source/Nevermore/Advanced/QueryBuilders/JoinSourceQueryBuilder.cs:41]: Not sure of the use of the `Join` method here, the `JoinSourceQueryBuilder` purpose seems to be towards setting up the `ON` statement, so the function and the additional `queryAlias` here serve no purpose, unless I am missing something.

Comment (tothegills) [reply source/Nevermore/Advanced/QueryBuilders/JoinSourceQueryBuilder.cs:41]: Yeah, this one looks a bit awkward, like the abstraction isn't quite right.

**Review by tothegills — APPROVED** [review]

**Pull request merged**